pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -221,6 +214,127 @@ func (pl *VolumeZone) Filter(ctx context.Context, cs *framework.CycleState, pod
+		return framework.QueueAfterBackoff, fmt.Errorf("unexpected objects in isSchedulableAfterStorageClassAdded: %w", err)
+	}
+
+	pvcNames := pl.getPersistentVolumeClaimNameFromPod(pod)
+	for _, pvcName := range pvcNames {
+		pvc, err := pl.pvcLister.PersistentVolumeClaims(pod.Namespace).Get(pvcName)
+		if err != nil {
+			logger.Error(err, "unexpected error during getting PVC", "Namespace", pod.Namespace, "pvc", pvcName)
+			continue
+		}
+		pvName := pvc.Spec.VolumeName
+		if pvName != "" {
+			// We can skip PersistentVolume because we only want to check storageClass
+			continue
+		}
+		scName := storagehelpers.GetPersistentVolumeClaimClass(pvc)
+		if scName == addedStorageClass.Name {
+			if (addedStorageClass.VolumeBindingMode != nil) && (*addedStorageClass.VolumeBindingMode == storage.VolumeBindingWaitForFirstConsumer) {
+				logger.V(5).Info("a new WaitForFirstConsumer storageClass for the pod's PVC is created, which might make the pod schedulable", "storageClass", klog.KObj(addedStorageClass), "pod", klog.KObj(pod))
+				return framework.QueueAfterBackoff, nil
+			}
+		}
+	}
+
+	logger.V(5).Info("StorageClass was created but it doesn't make this pod schedulable", "pod", klog.KObj(pod), "StorageClass", klog.KObj(addedStorageClass))
+	return framework.QueueSkip, nil
+}
+
+// isSchedulableAfterNodeChange is invoked whenever a node added or updated.
+// It checks whether the change of Node has made a previously unschedulable pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalNode, modifiedNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		return framework.QueueAfterBackoff, fmt.Errorf("unexpected objects in isSchedulableAfterNodeChange: %w", err)
+	}
+
+	podPVTopologies, status := pl.getPVbyPod(logger, pod)
+	if !status.IsSuccess() {
+		return framework.QueueAfterBackoff, nil
+	}
+	_, isMatched := pl.isSchedulableNode(logger, podPVTopologies, modifiedNode, pod)
+	if !isMatched {
+		logger.V(5).Info("node was created or updated but it doesn't make this pod schedulable", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+		return framework.QueueSkip, nil
+	}
+	wasMatched := false
+	if originalNode != nil {
+		_, wasMatched = pl.isSchedulableNode(logger, podPVTopologies, originalNode, pod)
+	}
+	if !wasMatched {
+		logger.V(5).Info("node was created or updated, which might make the pod schedulable", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+		return framework.QueueAfterBackoff, nil
+	}
+	logger.V(5).Info("node was created or updated but it doesn't make this pod schedulable", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
+	return framework.QueueSkip, nil
+}
+
+// isSchedulablerNode checks whether the node can schedule the pod.
+func (pl *VolumeZone) isSchedulableNode(logger klog.Logger, podPVTopologies []pvTopology, node *v1.Node, pod *v1.Pod) (*framework.Status, bool) {
```

Comment:
@sanposhiho , @kubernetes/sig-storage-misc 
In this function, if a Node has a TopologyZone label and the Pod has none, the Pod becomes schedulable on that Node. Is this logic correct?

Reply:
https://github.com/notpad/kubernetes/blob/4ff7251026011997e9a00c79b7ac0df00ae7ba30/pkg/scheduler/algorithm/predicates/predicates.go#L684-L697
> // NewVolumeZonePredicate evaluates if a pod can fit due to the volumes it requests, given
> // that some volumes may have zone scheduling constraints.  The requirement is that any
> // volume zone-labels must match the equivalent zone-labels on the node.  It is OK for
> // the node to have more zone-label constraints (for example, a hypothetical replicated
> // volume might allow region-wide access)

I couldn't find the KEP related to my question. However, I checked the previous code and found that the logic "if a Node has a TopologyZone label and the Pod has none, the Pod becomes schedulable on that Node" is valid.